### PR TITLE
Api client

### PR DESCRIPTION
**Pull request by maiamcc**
Client to talk to our API (so that we can easily stub functions out for test, change implementation of a certain query without affecting the entire bot, etc.)

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __init__(self, json: Dict):
+        self._json = json
+
+    @classmethod
+    def from_json(cls, json: Dict) -> 'Cop':
+        if 'name' not in json:
+            raise ValueError(f'Required field "name" not found in json: {json}')
+        cop = cls(json)
+        for k, v in json.items():
```
Comment (maiamcc) [inline api.py:22]: I swiped this initialization method from tweepy.model.Status. It's not super robust against changes to the api, unfortunately, and not great for type-hinting/auto-complete (see the awkward handful of fields specified at the top of the class) but it's good enough for government work I guess? (Happy to take suggestions on a better approach tho.)

File: api.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            if k == 'badge_number':
+                setattr(cop, k, int(v))
```
Comment (maiamcc) [inline api.py:24]: ...but also why is `badge_number` surfaced by the api/stored in the db as a string? 👀

File: app/api.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            response.raise_for_status()
+
+        resp_json = response.json()
+        cops = resp_json['cops']
+        pagination = resp_json['pagination_data']
+        logging.info(f'Got {len(cops)} hits (total: {pagination["total_count"]})')
+        return [Cop.from_json(cop) for cop in cops]
+
+    def _get(self, query_dict: Dict[str, str]):
+        # TODO: handle pagination, RIP.
```
Comment (mNov) [inline app/api.py:58]: heh, i'm not sure we ever implemented pagination on the api end - did we? i should take another look at that code.l

File: app/bot.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import logging
```
Comment (mNov) [inline app/bot.py:1]: all these app directory files were previously part of `app.py`, right? they look familiar

Comment (maiamcc) [reply app/bot.py:1]: correct!

File: app/handler.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    def _handle_tweet(self, body: str) -> str:
+        """Given tweet body, do any necessary API querying etc. and returns reply text."""
+        results: List[Cop] = []
+
+        if body == 'help':
+            return replies.HELP_REPLY
+        elif body.isdecimal():
+            results = self.cop_api_client.search(badge=int(body))
+        # TODO: if matches regex for (say) ([a-zA-Z.-]+) ([\d]+) (or whatever we
+        #   decide the syntax is), query by name AND badge number
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply app/handler.py:41]: whoop good call, will do

**Review by mNov — APPROVED** [review]
looks good!

**Pull request merged**